src/opm/parser/eclipse/EclipseState/Tables/TableManager.cpp

```diff
@@ -574,9 +599,79 @@ namespace Opm {
                 std::shared_ptr<RocktabTable> table = std::make_shared<RocktabTable>( dataItem , isDirectional, useStressOption );
                 container.addTable( tableIdx , table );
             }
+        }        
+    }
+    void TableManager::initRock2dTables(const Deck& deck) {
```

Comment:
Can the functions `TableManager::initRock2dTables()` and `TableManager::initRock2d()` be refactored to use a single implementation - I would think so?